samples/caf_sensor_manager/src/modules/workload_sim.c

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <caf/events/sensor_data_aggregator_event.h>
+
+#define BUSY_TIME CONFIG_WORKLOAD_SIM_BUSY_TIME
+
+
+static bool event_handler(const struct event_header *eh)
+{
+	if (is_sensor_data_aggregator_event(eh)) {
+		const struct sensor_data_aggregator_event *event =
+			cast_sensor_data_aggregator_event(eh);
+
+		/* Simulate processing of received data. */
+		k_busy_wait(BUSY_TIME * event->sample_cnt);
```

Comment:
This must not be done this way - we would block entire system as events are processed in workqueue.
Let's create a thread with preemptive priority and execute load from there.